app/routes.py

```diff
@@ -31,31 +32,52 @@ def create():
 def submit():
     submitForm = SubmitForm()
     form = LoginForm()
+    pathString = ""
+    challenge = getChallenge(int(submitForm.challenge_id.data)).toDict()
+    print(challenge)
+    pathString = challenge["startArticle"] + "|"
     for i in submitForm.path.data:
-        print(i)
-    return render_template('view.html', form=form, submitForm=submitForm, submitted=True)
+        if i.strip() != "":
+            pathString += i + "|"
+    pathString += challenge["endArticle"]
+    createNewSubmission(current_user.id, challenge["id"], pathString, int(time.time()))
+    return view(int(submitForm.challenge_id.data))
```

Comment:
This should be a redirect. bad practice and makes reloading the page re-send data and confuse the browser

Reply:
Should be updated